### DISCUSSION TITLE
Bipartite mappings with pytorch-geometric

### DISCUSSION THREAD

**Discussion by mishooax** [pyg-team/pytorch_geometric#5620] [General]
Hi,

Is there a pyg-tutorial example that shows how to create bipartite graph mappings between two data domains? I'm looking to recreate the model described in this paper: https://arxiv.org/abs/2202.07575. Specifically the mappings shown in Figure 2 of said paper:

<a href="https://ibb.co/nDMXZSt"><img src="https://i.ibb.co/wKhn31j/Screenshot-2022-10-06-at-20-39-11.png" alt="Screenshot-2022-10-06-at-20-39-11" border="0" /></a>

E.g. the "encoder" (leftmost graph) represents a mapping between the latitude/longitude data grid (blueish dots) and a coarser, icosahedron grid (gray nodes). Each gray node is connected to its 12 nearest neighbors on the lat/lon mesh.

I'm aware of this implementation: https://github.com/openclimatefix/graph_weather/blob/main/graph_weather/models/layers/encoder.py but i'm having trouble understanding some parts of the implementation and would like to work my way up from a simpler example, if possible.

Many thanks!

**Comment by rusty1s** [comment]
You can take a look at [`radius`](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.pool.radius) and [`knn`](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.pool.knn) functions in PyG, which can create a mapping between two sets of points. Let me know if this is helpful!

**Reply by mishooax** [reply]
Cheers - can I use these functions with the Haversine (great circle) distance? My graph nodes (coordinates == lat, lon in radians) lie on the sphere.
